benchalerts/parse_conbench.py

```diff
@@ -12,26 +12,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import textwrap
 from typing import List, Tuple
 
 
+def _clean(text: str) -> str:
+    """Clean text so it displays nicely as GitHub Markdown."""
+    return textwrap.fill(textwrap.dedent(text), 10000).replace("  ", "\n\n").strip()
+
+
 def benchmarks_with_z_regressions(all_comparisons: dict) -> List[Tuple[str, str]]:
-    """Find the run_ids and names of benchmarks whose z-scores were extreme enough to
-    constitute a regression.
+    """Find the compare_urls and names of benchmarks whose z-scores were extreme enough
+    to constitute a regression.
 
     Parameters
     ----------
     all_comparisons
-        The output of ConbenchClient.get_comparison_to_baseline().
+        The dict output of ConbenchClient.get_comparison_to_baseline().
 
     Returns
     -------
     List[Tuple[str, str]]
-        List of tuples of (run_id, benchmark_name) for all benchmarks with regressions.
+        List of tuples of (compare_url, benchmark_name) for all benchmarks with
+        regressions.
     """
     return [
-        (run_id, benchmark["benchmark"])
-        for run_id, benchmarks in all_comparisons.items()
+        (compare_url, benchmark["benchmark"])
+        for compare_url, benchmarks in all_comparisons.items()
         for benchmark in benchmarks
         if benchmark["contender_z_regression"]
     ]
+
+
+def regression_summary(
+    all_comparisons: dict, baseline_is_parent: bool, contender_sha: str
+) -> str:
+    """Generate a Markdown summary of what happened regarding regressions."""
+    regressions = benchmarks_with_z_regressions(all_comparisons)
+    summary = _clean(
+        f"""
+        Contender commit `{contender_sha[:8]}` had {len(regressions)} regressions
+        compared to its baseline commit.
+        """
+    )
+
+    if regressions:
+        summary += "\n\n### Benchmarks with regressions:"
+        previous_compare_url = ""
+        for compare_url, benchmark in regressions:
+            if compare_url != previous_compare_url:
+                run_id = compare_url[-32:]
+                summary += f"\n\n- Run ID [{run_id}]({compare_url})"
+                previous_compare_url = compare_url
+            summary += f"\n  - `{benchmark}`"
+
+    if not baseline_is_parent:
+        summary += "\n\n" + _clean(
+            """
+            ### Note
+
+            The baseline commit was not the immediate parent of the contender commit. If
+            this is a pull request, that's probably okay, because the baseline is the
+            commit from which your branch branched.
```

Comment:
We should have a todo at least to detect + fix this for the pull request type of comments. We don't have to do it on the first iteration, but generally speaking we shouldn't tell folks to ignore a warning

Reply:
Good idea. I added the option `warn_if_baseline_isnt_parent` so that when people implement `benchalerts` in their CI, they can turn off those warnings for pull request workflows but not for default branch workflows.